arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,389 @@
+        }
+    }
+    
+    // Convert the expression into a readable string.
+    func expressionToString(expression: AGSUtilityTraceConditionalExpression) -> String? {
+        if let categoryComparison = expression as? AGSUtilityCategoryComparison {
+            let comparisonOperatorString = categoryComparisonOperators[categoryComparison.comparisonOperator]
+            return "`\(categoryComparison.category.name)` \(comparisonOperatorString!)"
+        } else if let attributeComparison = expression as? AGSUtilityNetworkAttributeComparison {
+            // Check if attribute domain is a coded value domain.
+            if let domain = attributeComparison.networkAttribute.domain as? AGSCodedValueDomain {
+                // Get the coded value using the the attribute comparison value and attribute data type.
+                let dataType = attributeComparison.networkAttribute.dataType
+                let attributeValue = convertToDataType(otherValue: attributeComparison.value!, dataType: attributeComparison.networkAttribute.dataType)
+                let codedValue = domain.codedValues.first(where: { compare(dataType: dataType, comparee1: $0.code!, comparee2: attributeValue!) })
+                let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                return "'\(attributeComparison.networkAttribute.name)' \(comparisonOperatorString) '\(codedValue!.name)'"
+            } else {
+                if let nameOrValue = attributeComparison.otherNetworkAttribute?.name {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                } else if let nameOrValue = attributeComparison.value {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                }
+            }
+        } else if let andCondition = expression as? AGSUtilityTraceAndCondition {
+            return "(\(expressionToString(expression: andCondition.leftExpression)!)) AND\n(\(expressionToString(expression: andCondition.rightExpression)!))"
+        } else if let orCondition = expression as? AGSUtilityTraceOrCondition {
+            return "(\(expressionToString(expression: orCondition.leftExpression)!)) AND\n(\(expressionToString(expression: orCondition.rightExpression)!))"
+        }
+        return nil
+    }
+    
+    // Convert the values to matching data types.
+    func convertToDataType(otherValue: Any, dataType: AGSUtilityNetworkAttributeDataType) -> Any? {
+        switch dataType {
+        case .boolean:
+            return otherValue as! Bool
+        case .double:
+            if let valueString = otherValue as? String {
```

Comment:
When would this not be a `Double`? Same with the other cases that check for multiple types.

Reply:
I believe it depends on the type of attribute or value that was chosen. I adjusted the sample to allow negatives and decimals in which case could result in `Float`.